File: algorithms/LinearResearch.ts
{
    let n:number = 0
    let randomN:number = 0

    let inicialListNumber:number = 0
    let finalListNumber:number = 1000

    let attempts:number = 0

    let sucess:boolean = false

    const start = () => {

        //start
        const inicial = performance.now();
        randomValue()

        //loop
        while(sucess === false){
            
            attempt()
            attempts++

        }

        //sucess
        const end = performance.now();
        console.log("")
        console.log(`O algoritimo levou ${end - inicial} ms`);
        console.log(`Numero aleatorio entre ${finalListNumber} e 0 selecionado: ${randomN}`)
        console.log(`Numero de tentativas ${attempts}`);
        console.log("")

    }

    const randomValue = () => {

        randomN = Math.floor(Math.random() * (finalListNumber - inicialListNumber) + inicialListNumber)

    }

    const attempt = () => {
        if(n != randomN){
            n++
            console.log(n)
        }else{
            sucess = true
            console.log("")
            console.log("sucess")
            console.log("")
        }
    }

    start()
}

File: algorithms/LinearResearch/LinearResearch.py
from random import randint
import time

n = 0
randomN = 0

inicialListNumber = 0
finalListNumber = 100000000

attempts = 0

sucess = False

def start():
    
    global finalListNumber, n, randomN, attempts
    
    inicial = time.time()
    
    n = finalListNumber / 2
    randomValue()
    
    while(sucess == False):
        
        attempt()
        attempts = attempts + 1
        
    end = time.time()
    result = end - inicial
    print("")
    print("Numero aleatorio entre {0} e 0 selecionado: {1}".format(finalListNumber, randomN))
    print("O algoritimo levou {0}".format(result))
    print("Numero de tentativas {0}".format(attempts))
    print("")
    
def randomValue():
    
    global randomN
    
    randomN = randint(inicialListNumber, finalListNumber)
    
def attempt():
    
    global n, randomN, sucess
    
    if(n != randomN):
        n = n + 1
    else:
        sucess = True
        print("")
        print("sucess")
        print("")
        
start()